src/components/ProductCard/ProductCard.jsx

```diff
@@ -1,23 +1,73 @@
 import "./ProductCard.css";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts/index";
+import { addToWishlist, removeFromWishlist } from "../../services/index";
 
 export const ProductCard = ({ product }) => {
-  const {
-    _id,
-    productName,
-    price,
-    prevPrice,
-    imgSrc,
-    rating,
-    userSetQuantity,
-  } = { ...product };
+  const navigate = useNavigate();
 
+  const { userState, setUserState } = useAuth();
+  const { loginStatus, authToken, userDetails } = userState;
+  const { wishlist } = userDetails;
+
+  const { _id, productName, price, prevPrice, imgSrc, rating } = { ...product };
+
+  const [isWishlisted, setIsWishlisted] = useState({
+    value: false,
+    class: "not-wishlisted",
+  });
+
+  const isProductInWishlist = (wishlist, product) => {
+    const result = wishlist.filter((item) => item._id === product._id);
```

Comment:
Read about `some` funciton. It makes this functionality easier.

Reply:
yes it did reduce my work, changes done